lib/std/priority_dequeue.zig

```diff
@@ -21,39 +21,41 @@ pub fn PriorityDequeue(comptime T: type, comptime Context: type, comptime compar
 
         items: []T,
         len: usize,
-        allocator: Allocator,
         context: Context,
 
         /// Initialize and return a new priority dequeue.
-        pub fn init(allocator: Allocator, context: Context) Self {
+        pub fn init(context: Context) Self {
             return Self{
                 .items = &[_]T{},
                 .len = 0,
-                .allocator = allocator,
                 .context = context,
             };
         }
 
         /// Free memory used by the dequeue.
-        pub fn deinit(self: Self) void {
-            self.allocator.free(self.items);
+        pub fn deinit(self: Self, allocator: std.mem.Allocator) void {
+            if (self.items.len > 0) {
```

Comment:
This isn't strictly necessary since `free` checks for 0-length slices but it still seemed odd to me to intentionally free something that was not a result of a call to `alloc` so I added this check.